zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -410,6 +415,10 @@ angular.module('zeppelinWebApp')
 
   $scope.aceChanged = function() {
     $scope.dirtyText = $scope.editor.getSession().getValue();
+
+    $timeout(function() {
+      $scope.setParagraphMode($scope.editor.getSession(), $scope.editor.getCursorPosition());
+    });
```

Comment:
I haven't used Angular.js for some time and I don't quite remember its APIs but shouldn't `$timeout` accept a second parameter - the duration?